backend/app/api/auterity_expansion.py

```
@@ -370,165 +387,83 @@ async def deploy_agent(
-            detail=f"Agent deployment failed: {str(e)}",
+            detail=f"Failed to deploy agent: {str(e)}",
         )
 
 
 @router.post("/agents/{agent_id}/tasks")
-async def assign_task(
+async def assign_task_to_agent(
     agent_id: UUID,
-    task_data: dict,
+    task_data: Dict[str, Any],
     db: Session = Depends(get_db),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)